kratos/utilities/string_utilities.cpp

```diff
@@ -175,4 +175,52 @@ std::string ReplaceAllSubstrings(
     return output_string;
 }
 
+std::string Trim(
+    const std::string& rInputString,
+    const bool RemoveNullChar)
+{
+    return TrimLeft(TrimRight(rInputString, RemoveNullChar), RemoveNullChar);
+}
+
+
+const auto trim_space_and_null_char = [](std::string::value_type character) -> bool {
+    return std::isspace(character) || character == '\0';
+};
+
+const auto trim_space = [](std::string::value_type character) -> bool {
+    return std::isspace(character);
+};
+
+std::string TrimLeft(
+    const std::string& rInputString,
+    const bool RemoveNullChar)
+{
+    std::string output_string(rInputString);
+
+    const auto trim_char = RemoveNullChar ? trim_space_and_null_char : trim_space;
+
+    output_string.erase(output_string.begin(), std::find_if(output_string.begin(), output_string.end(),
+            [&trim_char](std::string::value_type ch) {return !trim_char(ch);}
+        )
+    );
```

Comment:
```suggestion
    if (RemoveNullChar) {
        output_string.erase(output_string.begin(), std::find_if(output_string.begin(), output_string.end(),
                [](std::string::value_type ch) {return !TrimSpaceAndNullChar(ch);}
            )
        );
    } else {
        output_string.erase(output_string.begin(), std::find_if(output_string.begin(), output_string.end(),
                [](std::string::value_type ch) {return !TrimSpace(ch);}
            )
        );
    }
```

Reply:
that is exactly what I wanted to avoid :sweat_smile: 

I think I will revert back to the `std::function` solution. Performance does not matter here anyway

Reply:
alright